packages-content-model/roosterjs-content-model-editor/lib/modelApi/table/applyTableFormat.ts

```diff
@@ -66,29 +70,32 @@ function clearCache(rows: ContentModelTableRow[]) {
     });
 }
 
-function updateBgColorOverrides(rows: ContentModelTableRow[], forceClear: boolean): boolean[][] {
-    const result: boolean[][] = [];
+function updateOverrides(rows: ContentModelTableRow[], forceClear: boolean): MetaOverrides {
+    const overrides: MetaOverrides = { bgColorOverrides: [], vAlignOverrides: [] };
 
     rows.forEach(row => {
-        const currentRow: boolean[] = [];
+        const bgColorOverrides: boolean[] = [];
+        const vAlignOverrides: boolean[] = [];
 
-        result.push(currentRow);
+        overrides.bgColorOverrides.push(bgColorOverrides);
+        overrides.vAlignOverrides.push(vAlignOverrides);
 
         row.cells.forEach(cell => {
             updateTableCellMetadata(cell, metadata => {
                 if (metadata && forceClear) {
-                    currentRow.push(false);
+                    bgColorOverrides.push(false);
                     delete metadata.bgColorOverride;
                 } else {
-                    currentRow.push(!!metadata?.bgColorOverride);
+                    bgColorOverrides.push(!!metadata?.bgColorOverride);
                 }
+                vAlignOverrides.push(!!metadata?.vAlignOverride);
```

Comment:
Question: do we also need to delete metadata.verticalAlign when forceClear is true?

Reply:
Ok, in that case we should rename this parameter to reflect the fact that it will only impact sharde